### DISCUSSION TITLE
React on x-intersect:leave only after element has been visible

### DISCUSSION THREAD

**Discussion by simonferndriger** [alpinejs/alpine#4016] [1. Help] [answered]
When I use [x-intersect:leave](https://alpinejs.dev/plugins/intersect#x-intersect-leave) it triggers an event for elements below the fold that are initially not visible. What is the best way to be able to react only when the element has been visible and *then* leaves the viewport?

**Comment by ekwoka** [comment, marked as answer]
The simplest userland version is to have something that tracks whether the whole thing has "initialized" and only start reacting afterwards.

or ensure your handler for it is idempotent to begin with. Is there a reason that it's an issue as is?

https://developer.mozilla.org/en-US/docs/Web/API/IntersectionObserver/observe

the behavior is standard to the underlying API.

**Reply by simonferndriger** [reply]
My use case is that I remove ads that have been scrolled by to improve page responsivness. But they need to show up first ;)

**Reply by ekwoka** [reply]
Then just a little code to skip the first time it's run would do fine.

Or on a parent tracking an `init` that runs after all the directives would have been evaluated for the first time.

**Reply by simonferndriger** [reply]
I set `initialized` to `true`  after the first `x-intersect`
